client/src/components/PassReset.tsx

```diff
@@ -0,0 +1,187 @@
+    e.preventDefault(); // no refresh; default
+
+    const msg = getErrorMessages();
+    if (msg !== "") {
+      console.warn(msg);
+      props.showAlert("warning", msg);
+      return;
+    }
+
+    if (isFormMissingFields()) {
+      const msg = "Please fill in all the fields.";
+      console.warn(msg);
+      props.showAlert("warning", msg);
+      return;
+    }
+
+    try {
+      const { status, data } = await api.sendPassReset(passResetForm);
+      if (status !== 204) {
+        if (!data) {
+          throw new Error("Missing error response.");
+        }
+        const msg = data.message;
+        console.error(msg);
+        props.showAlert("error", msg);
+      } else {
+        props.showAlert("success", "Password reset email sent!")
+      }
+    } catch (error) {
+      console.error(error);
+      props.showAlert(
+        "error",
+        "An error occurred logging you in. Please try again later."
+      );
+    }
+  };
+
+  const handleFormInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPassResetForm({ ...passResetForm, [e.target.name]: e.target.value });
+  };
+
+  const handleError =
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>) =>
+    (msg: string) => {
+      return setPassResetFormErrors({
+        ...passResetFormErrors,
+        [e.target.name]: msg,
+      });
+    };
+
+  return (
+    <Box
+      data-testid="ResetPassTab"
+      sx={{
+        mx: 2,
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+      }}
+      overflow="auto"
+      maxHeight="70vh"
+    >
+      <Avatar sx={{ m: 1, bgcolor: "secondary.main" }}>
+        <LockOutlinedIcon />
+      </Avatar>
+      <Typography component="h1" variant="h5">
+        Reset Password
+      </Typography>
+      <Box
+        component="form"
+        noValidate
+        autoComplete="off"
+        onSubmit={handleSubmit}
+        data-testid="form"
+        sx={{ mt: 1 }}
+      >
+        <TextField
+          name="email"
+          label="Email Address"
+          onChange={handleFormInput}
+          fullWidth
+          required
+          data-testid="resetField"
+          type="email"
+          onBlur={(e) =>
+            validateBlur(
+              /..*@(mail\.|alum\.){0,}utoronto.ca$/,
+              passResetForm.email,
+              handleError(e),
+              "Invalid email, only utoronto emails allowed"
+            )
+          }
+          error={passResetFormErrors.email !== ""}
+          helperText={passResetFormErrors.email}
+        />
+        <Button
+          type="submit"
+          variant="contained"
+          fullWidth
+          sx={{ mt: 3, mb: 2 }}
+        >
+          Reset my password
+        </Button>
+        <Grid container>
+          <Grid item>
+            <Link
+              href="#"
+              variant="body2"
+              onClick={(e) => props.handleChange(e, 0)}
```

Comment:
I'm assuming that the `0` here is the index? If so, this should be passed in, or handleChange should only take an error and the container which uses this component should worry about the index

Reply:
This index is for the TabPanel, and it works just like how the SignIn and SignUp components do for switching between tabs.